backend/test/integration/artifact_api_test.go

```diff
@@ -0,0 +1,300 @@
+		ID: experimentID,
+	}); err != nil {
+		s.T().Logf("Failed to clean up test experiment %s: %v", experimentID, err)
+	}
+}
+
+func (s *ArtifactAPITest) deleteRun(runID string) {
+	if err := s.runClient.Delete(&runParams.RunServiceDeleteRunV1Params{
+		ID: runID,
+	}); err != nil {
+		s.T().Logf("Failed to clean up test run %s: %v", runID, err)
+	}
+}
+
+func (s *ArtifactAPITest) createExperiment() *experiment_model.APIExperiment {
+	experimentName := fmt.Sprintf("artifact-test-experiment-%d", time.Now().Unix())
+	experiment := test.GetExperiment(experimentName, "Test for artifact reading", s.namespace)
+	experiment, err := s.experimentClient.Create(&params.ExperimentServiceCreateExperimentV1Params{Experiment: experiment})
+	require.NoError(s.T(), err)
+	return experiment
+}
+
+func (s *ArtifactAPITest) uploadPipeline() *pipeline_upload_model.APIPipeline {
+	pipelineParams := uploadParams.NewUploadPipelineParams()
+	pipelineName := fmt.Sprintf("large-artifact-test-%d", time.Now().Unix())
+	pipelineParams.SetName(&pipelineName)
+	pipeline, err := s.pipelineUploadClient.UploadFile("../resources/large_artifact_pipeline.yaml", pipelineParams)
+	require.NoError(s.T(), err, "Failed to upload pipeline")
+	return pipeline
+}
+
+func (s *ArtifactAPITest) runPipeline() (runID, experimentID, pipelineID string) {
+	experiment := s.createExperiment()
+	pipeline := s.uploadPipeline()
+
+	runRequest := &runParams.RunServiceCreateRunV1Params{Run: &run_model.APIRun{
+		Name: fmt.Sprintf("test-artifact-run-%d", time.Now().Unix()),
+		PipelineSpec: &run_model.APIPipelineSpec{
+			PipelineID: pipeline.ID,
+		},
+		ResourceReferences: []*run_model.APIResourceReference{
+			{
+				Key: &run_model.APIResourceKey{
+					Type: run_model.APIResourceTypeEXPERIMENT.Pointer(),
+					ID:   experiment.ID,
+				},
+				Relationship: run_model.APIRelationshipOWNER.Pointer(),
+			},
+		},
+	}}
+	run, _, err := s.runClient.Create(runRequest)
+	require.NoError(s.T(), err)
+	return run.Run.ID, experiment.ID, pipeline.ID
+}
+
+func (s *ArtifactAPITest) waitForRunCompletion(runID string) {
+	t := s.T()
+	maxWaitTime := 2 * time.Minute
+	startTime := time.Now()
+
+	for time.Since(startTime) < maxWaitTime {
+		runDetail, _, err := s.runClient.Get(&runParams.RunServiceGetRunV1Params{RunID: runID})
+		require.NoError(t, err)
+
+		if runDetail.Run.Status == "Succeeded" || runDetail.Run.Status == "Completed" ||
+			runDetail.Run.Status == "Failed" || runDetail.Run.Status == "Error" {
+			require.Contains(t, []string{"Succeeded", "Completed"}, runDetail.Run.Status,
+				"Run should have succeeded")
+			t.Log("Run completed")
+			return
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+
+	require.Fail(t, "Run did not complete within %v", maxWaitTime)
+}
+
+func (s *ArtifactAPITest) extractWorkflowNodeID(runID string) string {
+	t := s.T()
+
+	resp, err := http.Get(fmt.Sprintf("http://localhost:8888/apis/v1beta1/runs/%s", runID))
```

Comment:
can you re-use the api server host info  (as well as the scheme http/https) from `backend/test/config/flags.go` instead of hardcoding it? This applies to everywhere, this applies to any place you are hardcoding localhost:8888